internal/serverinstall/nomad.go

```diff
@@ -319,11 +332,11 @@ func (i *NomadInstaller) Upgrade(
 	sg := ui.StepGroup()
 	defer sg.Wait()
 
-	if !i.config.consulService {
+	if i.config.serviceProvider == "none" && !i.config.consulService {
 		// By default, we don't auto-enable the consul service because prior to Waypoint
 		// version 0.6.2, we did not enable it by default.
-		sw := sg.Add("Nomad Consul Service is disabled. If you had previously enabled " +
-			"it in the last installation, please stop this upgrade and re-run with -nomad-consul-service=true.")
+		sw := sg.Add("Service discovery is disabled for the Waypoint Nomad job. If you had previously enabled " +
+			"it in the last installation, please stop this upgrade and re-run with -nomad-service-provider flag.")
```

Comment:
@demophoon after giving this some thought, I think an interactive terminal event could be useful here. If the user is OK with proceeding, we let them, otherwise they can abort!

```suggestion
		proceed, err := opts.UI.Input(&terminal.Input{
			Prompt: "Service discovery is disabled for the Waypoint Nomad job. If you had previously enabled " +
				"it in the last installation, please stop this upgrade and re-run with the -nomad-service-provider flag. " +
				"Otherwise, enter 'yes' to continue the upgrade: ",
			Style:  "",
			Secret: false,
		})
		if err != nil {
			opts.UI.Output(
				"Error uninstalling server: %s",
				clierrors.Humanize(err),
				terminal.WithErrorStyle(),
			)
		} else if strings.ToLower(proceed) != "yes" {
			return nil, err
		}
```

Reply:
This is a good solution to this particular concern!